### DISCUSSION TITLE
`TabView` how to detect new tab becomes visible in the middle of switching

### DISCUSSION THREAD

**Discussion by adxzhang1** [react-navigation/react-navigation#12308] [Q&A]
When using `TabView`, how do I detect if a new tab becomes visible while user is in the middle of switching tabs? I want to perform some logic as the new tab becomes visible. `onIndexChange` fires only after the new tab is fully switched to.

Ex: I have 2 tabs: Tab1 & Tab2. User starts on Tab1. User starts panning to Tab2 and they are able to partially see Tab2.

**Comment by satya164** [comment]
You get a `position` animated value in the props of your scene. You can add a listener to it and compare it with the offset of the screen.

Depending on the use case, you may also use `renderLazyPlaceholder` to render placeholder content that'll be replaced by actual content when screen comes into view.
